moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -326,48 +285,43 @@ struct DistanceResultsData
     clear();
   }
 
-  /// The minimum distance between two objects. If two objects are in collision, min_distance <= 0.
-  double min_distance;
+  /// The distance between two objects. If two objects are in collision, distance <= 0.
+  double distance;
 
   /// The nearest points
   Eigen::Vector3d nearest_points[2];
 
   /// The object link names
-  std::string link_name[2];
-
-  /// The gradient
-  Eigen::Vector3d gradient;
+  std::string link_names[2];
 
-  /// Indicates if gradient was calculated.
-  bool hasGradient;
+  /// A normalized vector pointing from link_names[0] to link_names[1].
```

Comment:
Shouldn't that read:
> A normalized vector pointing from nearest_points[0] to nearest_points[1]

Reply:
Yep

Reply:
Sorry, I believe it should be link_names[0] to link_names[1] mainly because in the event of a collision the nearest_points are the same but the normal is always calculate such that it points from link_names[0] to link_names[1].